tests/unit/amo/components/TestSearchFilters.js

```diff
@@ -386,6 +387,45 @@ describe(__filename, () => {
         }),
       });
     });
+
+    it('does not pass sort=random when the recommended checkbox is unchecked', () => {
+      const root = render({
+        _config,
+        filters: {
+          recommended: true,
+          sort: SEARCH_SORT_RANDOM,
+        },
+      });
+
+      const checkbox = root.find('.SearchFilters-Recommended');
+      checkbox.simulate('change', createFakeEvent());
+
+      sinon.assert.calledWithExactly(fakeHistory.push, {
+        pathname: `/en-US/android/search/`,
+        query: convertFiltersToQueryParams({}),
+      });
+    });
+
+    it('does pass sort=random when the recommended checkbox is checked', () => {
+      const root = render({
+        _config,
+        filters: {
+          recommended: false,
+          sort: SEARCH_SORT_RANDOM,
```

Comment:
Hmm, this is an interesting test because our intent is that this state could never exist. That is, having `recommended=false` and `sort=random` together is supposed to be illegal. So maybe we shouldn't have this test at all?

Reply:
Indeed. That might not be needed.